### PR TITLE
clone settings import settings

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/core/models/misc/expense-field.model.ts [hunk 1]
```diff
@@ -2,3 +2,12 @@ export type ExpenseField = {
   attribute_type: string;
   display_name: string;
 };
+
+export type ExpenseFieldFormArray = {
+  source_field: string;
+  destination_field: string;
+  import_to_fyle: boolean;
+  disable_import_to_fyle: boolean;
+  source_placeholder: string,
```
Comment (ashwin1111) [inline src/app/core/models/misc/expense-field.model.ts:11]: this is same as `ExpenseFieldsFormOption`. Can extend this model and add just `addSourceField`

File: src/app/core/services/misc/mapping.service.ts [hunk 2]
```diff
@@ -216,4 +216,7 @@ export class MappingService {
     return undefined;
   }
 
+  getQboField(): Observable<ExpenseField[]> {
+    return this.apiService.get(`/workspaces/${this.workspaceId}/qbo/qbo_fields/`, {});
+  }
```
Comment (ashwin1111) [inline src/app/core/services/misc/mapping.service.ts:221]: where is this used?

File: src/app/integration/onboarding/clone-settings/clone-settings.component.html [hunk 3]
```diff
@@ -360,6 +361,256 @@ <h5 class="clone-settings--header-caption sub-text-color">In this section, you c
...
+            <div>
+                <div fxLayoutAlign="start start" class="clone-settings--header-section"
+                    data-cy="configuration-contents">
+                    <div class="clone-settings--header-icon">
+                        <img src="assets/images/svgs/general/setting.svg">
+                    </div>
+                    <div>
+                        <h3>Import Settings</h3>
+                        <h5 class="clone-settings--header-caption sub-text-color">In this section, you can configure how
+                            the dimensions you import from Quickbooks should be mapped in Fyle</h5>
```
Comment (ashwin1111) [inline src/app/integration/onboarding/clone-settings/clone-settings.component.html:374]: QuickBooks Online

File: src/app/integration/onboarding/clone-settings/clone-settings.component.html [hunk 4]
```diff
@@ -360,6 +361,256 @@ <h5 class="clone-settings--header-caption sub-text-color">In this section, you c
...
+                            <img class="clone-settings--info-icon" src="assets/images/svgs/general/info-disabled.svg"
+                                matTooltip="Vendors from QuickBooks Online will be imported as merchants in Fyle and will be a selectable field while creating an expense."
+                                matTooltipPosition="above" matTooltipClass="clone-settings">
+                        </div>
+
+                        <div>
+                            <app-toggle 
+                                [isCloneSettings]="true" 
+                                [form]="cloneSettingsForm"
+                                [formControllerName]="'importItems'">
```
Comment (ashwin1111) [inline src/app/integration/onboarding/clone-settings/clone-settings.component.html:415]: controller name is incorrect

File: src/app/integration/onboarding/clone-settings/clone-settings.component.html [hunk 5]
```diff
@@ -360,6 +361,256 @@ <h5 class="clone-settings--header-caption sub-text-color">In this section, you c
...
+                            </app-toggle>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="clone-settings--field-header-section">
+                    <div fxLayoutAlign="start center" class="clone-settings--field-header">
+                        <div class="clone-settings--qbo-header">
+                            Quickbooks Field
```
Comment (ashwin1111) [inline src/app/integration/onboarding/clone-settings/clone-settings.component.html:424]: QuickBooks Online

File: src/app/integration/onboarding/clone-settings/clone-settings.component.ts [hunk 6]
```diff
@@ -327,7 +436,16 @@ export class CloneSettingsComponent implements OnInit {
...
+      searchOption: [],
+
+      // Import Settings
+      chartOfAccount: [this.cloneSettings.import_settings.workspace_general_settings.import_categories],
+      importItems: [this.cloneSettings.import_settings.workspace_general_settings.import_items],
+      chartOfAccountTypes: this.formBuilder.array(chartOfAccountTypeFormArray),
+      expenseFields: this.formBuilder.array(expenseFieldsFormArray),
+      taxCode: [this.cloneSettings.import_settings.workspace_general_settings.import_tax_codes],
+      defaultTaxCode: [this.cloneSettings.import_settings.general_mappings?.default_tax_code?.id ? this.cloneSettings.import_settings.general_mappings.default_tax_code : null],
+      importVendorsAsMerchants: [this.cloneSettings.import_settings.workspace_general_settings.import_vendors_as_merchants]
```
Comment (ashwin1111) [inline src/app/integration/onboarding/clone-settings/clone-settings.component.ts:448]: I don't see a field in UI for `importVendorsAsMerchants`. Please help me locate if it's already present.

File: src/app/integration/onboarding/clone-settings/clone-settings.component.ts [hunk 7]
```diff
@@ -337,16 +455,71 @@ export class CloneSettingsComponent implements OnInit {
...
   private setupPage(): void {
     const destinationAttributes = ['BANK_ACCOUNT', 'CREDIT_CARD_ACCOUNT', 'ACCOUNTS_PAYABLE', 'VENDOR'];
 
     forkJoin([
       this.cloneSettingService.getCloneSettings(),
       this.mappingService.getGroupedQBODestinationAttributes(destinationAttributes),
-      this.mappingService.getMappingSettings()
+      this.mappingService.getMappingSettings(),
+      this.mappingService.getFyleExpenseFields(),
+      this.mappingService.getQBODestinationAttributes('TAX_CODE')
```
Comment (ashwin1111) [inline src/app/integration/onboarding/clone-settings/clone-settings.component.ts:519]: use enum, don't hard code string

File: src/app/core/services/misc/mapping.service.ts (hunk 2 above)
Comment (NileshPant1999) [reply src/app/core/services/misc/mapping.service.ts:221]: removing it

Comment (NileshPant1999) [reply src/app/core/services/misc/mapping.service.ts:221]: i think i was using this earlier but later found of no use, removing this

File: src/app/integration/onboarding/clone-settings/clone-settings.component.ts (hunk 6 above)
Comment (NileshPant1999) [reply src/app/integration/onboarding/clone-settings/clone-settings.component.ts:404]: i forget to replace formControlName in the UI, it is changed now

File: src/app/core/models/misc/expense-field.model.ts (hunk 1 above)
Comment (NileshPant1999) [reply src/app/core/models/misc/expense-field.model.ts:11]: we can use ExpenseFieldsFormOption only,  no need for this

Comment (NileshPant1999) [reply src/app/core/models/misc/expense-field.model.ts:11]: just add a green here, i will remove this off in the next commit

Comment (ashwin1111) [reply src/app/core/models/misc/expense-field.model.ts:11]: sure

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**